src/predict_request_validation_utils.cpp

```diff
@@ -0,0 +1,301 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "predict_request_validation_utils.hpp"
+
+#include <sstream>
+#include <string>
+
+#include <spdlog/spdlog.h>
+
+#include "modelconfig.hpp"
+
+namespace ovms {
+namespace validation_utils {
+
+Status validateNumberOfInputs(const tensorflow::serving::PredictRequest& request, const size_t expectedNumberOfInputs) {
+    if (request.inputs_size() > 0 && expectedNumberOfInputs == static_cast<size_t>(request.inputs_size())) {
+        return StatusCode::OK;
+    }
+    std::stringstream ss;
+    ss << "Expected: " << expectedNumberOfInputs << "; Actual: " << request.inputs_size();
+    const std::string details = ss.str();
+    SPDLOG_DEBUG("[requested endpoint:{} version:{}] Invalid number of inputs - {}", request.model_spec().name(), request.model_spec().version().value(), details);
```

Comment:
```suggestion
    SPDLOG_DEBUG("[requested servable:{} version:{}] Invalid number of inputs: {}", request.model_spec().name(), request.model_spec().version().value(), details);
```

Reply:
I agree with servable word, but do not agree with removal of `-`. This nicely splits log into chunks where everything after `-` is sent to the client.